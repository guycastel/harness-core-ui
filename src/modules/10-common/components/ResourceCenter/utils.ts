/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import moment from 'moment'
import type { AccountDTO, UserInfo } from 'services/cd-ng'

export const ON_PREM_RELEASE_NOTE_LINK = 'https://ngdocs.harness.io/article/556wy85kbo-harness-on-prem-release-notes'

export const HARNESS_SEARCH_LINK = 'https://harness.io/search/'
export const HARNESS_UNIVERISITY_LINK = 'https://developer.harness.io/certifications/'
export const HARNESS_COMMUNITY_LINK = 'https://community.harness.io/'
export const HARNESS_COMMUNITY_SLACK_LINK =
  'https://join.slack.com/t/harnesscommunity/shared_invite/zt-y4hdqh7p-RVuEQyIl5Hcx4Ck8VCvzBw'
export const HARNESS_DOCS_LINK = 'https://docs.harness.io/'
export const HARNESS_API_DOCS_LINK = 'https://apidocs.harness.io/'
export const HARNESS_TUTORIALS = 'https://developer.harness.io/tutorials/get-started'
export const SITE_STATUS_LINK = 'https://status.harness.io/'
export const HARNESS_JFROG =
  'https://developer.harness.io/docs/continuous-integration/use-ci/build-and-upload-artifacts/upload-artifacts-to-jfrog'
export const HARNESS_SUPPORT_PAGE = 'https://www.harness.io/support-search'

export const timestamp = moment.now()
export const HARNESS_SUPPORT_LINK =
  '/sso.html?action=login&brand_id=114095000394&locale_id=1&return_to=https%3A%2F%2Fsupport.harness.io%2Fhc%2Fen-us%2Frequests&src=zendesk&timestamp=' +
  timestamp
export const CANNY_SUPPORT_LINK = 'https://ideas.harness.io/'
export const WHATS_NEW_LINK = `https://docs.harness.io/article/ueeiji09do-what-s-new`
export const EARLY_ACCESS_LINK = `https://docs.harness.io/article/w4krvu96i3-early-access`

export const openZendeskSupport = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(HARNESS_SUPPORT_LINK)
}

const getInitials = (name: string) => {
  const parts = name.split('.')
  let initials = ''
  for (let i = 0; i < parts.length; i++) {
    if (parts[i].length > 0 && parts[i] !== '') {
      initials += parts[i][0]
    }
  }
  return initials
}
const initCanny = (currentUserInfo: UserInfo, account?: AccountDTO) => {
  Canny('identify', {
    appID: window.cannyAppId,
    user: {
      companies: [
        {
          id: account?.identifier,
          name: account?.companyName
        }
      ],
      email: currentUserInfo.email,
      id: currentUserInfo.uuid,
      name: account?.cannyUsernameAbbreviationEnabled
        ? currentUserInfo.name?.includes('@')
          ? getInitials(currentUserInfo.name.split('@')[0] || '') || ''
          : getInitials(currentUserInfo.name || '') || ''
        : currentUserInfo.name || ''
    }
  })
}

export const cannySupportShareYourIdeas = (
  e: React.MouseEvent<Element, MouseEvent>,
  currentUserInfo: UserInfo,
  account?: AccountDTO
): void => {
  e.stopPropagation()
  e.preventDefault()
  initCanny(currentUserInfo, account)
  window.open(CANNY_SUPPORT_LINK)
}

export const openWhatsNew = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(WHATS_NEW_LINK)
}

export const openEarlyAccess = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(EARLY_ACCESS_LINK)
}

export const getReleaseNotesLink = () => {
  return window.releaseNotesLink || ON_PREM_RELEASE_NOTE_LINK
}
