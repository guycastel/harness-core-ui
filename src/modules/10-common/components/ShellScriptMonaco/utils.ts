/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

const EXPRESSION_START_REGEX = /<\+([A-Za-z0-9_.'"()]*?)$/

export function getStartColumnForMonacoRange(prevText?: string): number | undefined {
  if (typeof prevText !== 'string') return undefined
  const startColumnWithTriggerChar = prevText.match(EXPRESSION_START_REGEX)?.index
  if (startColumnWithTriggerChar === undefined) return undefined
  const startColumn = startColumnWithTriggerChar + 2 + 1 // 2 for trigger chars + 1 for monaco as it starts index from 1
  return startColumn
}
