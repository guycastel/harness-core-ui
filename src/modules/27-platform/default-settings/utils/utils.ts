/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { StoreMetadata, StoreType } from '@common/constants/GitSyncTypes'
import { SettingType } from '@common/constants/Utils'
import { ResponseListSettingResponseDTO } from 'services/cd-ng'

export const AidaAgreementType = 'AIDA'
export const defaultSettingsTagInputSeparator = ','

export const getSettingValue = (
  settingsData: ResponseListSettingResponseDTO | null,
  identifier: string
): string | undefined => {
  let settingValue: string | undefined

  if (settingsData?.data?.length) {
    settingValue = settingsData?.data?.find(element => element.setting?.identifier === identifier)?.setting?.value
  }
  return settingValue
}

export const getDefaultStoreType = (
  settingsData: ResponseListSettingResponseDTO | null
): StoreMetadata['storeType'] => {
  const settingValue = getSettingValue(settingsData, SettingType.DEFAULT_STORE_TYPE_FOR_ENTITIES)
  return settingValue === StoreType.REMOTE ? StoreType.REMOTE : StoreType.INLINE
}

/**
 * This method will include both ENFORCE_GIT_EXPERIENCE and DEFAULT_STORE_TYPE_FOR_ENTITIES settings
 * INLINE is used as fallback if no gitXSetting is defined
 * @param gitXSetting
 * @returns
 */
export function getDefaultStoreTypeFromSettings(
  gitXSetting?: ResponseListSettingResponseDTO | null
): StoreMetadata['storeType'] {
  return gitXSetting
    ? getSettingValue(gitXSetting, SettingType.ENFORCE_GIT_EXPERIENCE) === 'true'
      ? StoreType.REMOTE
      : getDefaultStoreType(gitXSetting)
    : StoreType.INLINE
}
