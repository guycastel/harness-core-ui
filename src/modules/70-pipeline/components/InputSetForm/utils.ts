/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { SaveToGitFormInterface } from '@modules/10-common/components/SaveToGitForm/SaveToGitForm'
import type { EntityGitDetails } from 'services/pipeline-ng'

export interface GetUpdatedGitDetailsReturnType extends EntityGitDetails {
  lastObjectId?: string
  lastCommitId?: string
  baseBranch?: string
}

export const getUpdatedGitDetails = (
  isEdit: boolean,
  gitDetails: SaveToGitFormInterface | undefined,
  lastObjectId: string,
  initialGitDetails: EntityGitDetails,
  conflictCommitId?: string
): GetUpdatedGitDetailsReturnType => {
  let updatedGitDetails: GetUpdatedGitDetailsReturnType = {}
  if (gitDetails) {
    updatedGitDetails = { ...gitDetails }
    if (isEdit) {
      updatedGitDetails['lastObjectId'] = lastObjectId
      updatedGitDetails['lastCommitId'] = conflictCommitId || initialGitDetails.commitId
    }
    if (gitDetails.isNewBranch) {
      updatedGitDetails['baseBranch'] = initialGitDetails.branch
    }
  }
  return updatedGitDetails
}
