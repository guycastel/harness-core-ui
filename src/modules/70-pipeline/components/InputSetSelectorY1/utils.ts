/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { SelectedInputSetListValue } from '../InputSetSelector/SelectedInputSetList'

export const INPUT_SET_SELECTOR_PAGE_SIZE = 100

export const isInputSetItemSelected = (selected: SelectedInputSetListValue[], identifier: string): boolean => {
  return !!selected.find(item => item.value === identifier)
}
