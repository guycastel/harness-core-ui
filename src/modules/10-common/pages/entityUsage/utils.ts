/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { useMemo } from 'react'
import { useParams } from 'react-router-dom'
import { MultiSelectOption } from '@harness/uicore'
import { ProjectPathProps, SecretsPathProps } from '@common/interfaces/RouteInterfaces'
import { ListActivitiesQueryParams } from 'services/cd-ng'
import { StringKeys } from 'framework/strings'
import { COMMON_PAGE_SIZE_OPTIONS } from '@modules/10-common/constants/Pagination'
import { getWindowLocationUrl } from 'framework/utils/WindowLocation'
import { EntityType } from './EntityConstants'

export const pageSize = COMMON_PAGE_SIZE_OPTIONS[1]

enum FilterScope {
  ACCOUNT = 'account',
  ORG = 'org',
  PROJECT = 'project'
}
export const getScopeSelectOptions = (getString: (key: StringKeys) => string): MultiSelectOption[] => {
  const scopeList = [
    { label: getString('account'), value: FilterScope.ACCOUNT },
    { label: getString('orgLabel'), value: FilterScope.ORG },
    { label: getString('projectLabel'), value: FilterScope.PROJECT }
  ]
  return scopeList
}

export const useSecretRuntimeUsageTimeData = (): { threeMonthsBack: Date; currentDate: number } => {
  return useMemo(() => {
    const threeMonthsBack = new Date()
    threeMonthsBack.setMonth(threeMonthsBack.getMonth() - 3)
    const currentDate = Date.now()

    return {
      threeMonthsBack: threeMonthsBack,
      currentDate: currentDate
    }
  }, [])
}

export const useGetSecretRuntimeUsageQueryParams = (
  page: number,
  searchTerm: string | undefined,
  pageContentSize?: number
): ListActivitiesQueryParams => {
  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
  const { secretId } = useParams<SecretsPathProps>()
  return {
    accountIdentifier: accountId,
    projectIdentifier: projectIdentifier,
    orgIdentifier: orgIdentifier,
    identifier: secretId,
    pageIndex: page,
    pageSize: pageContentSize || pageSize,
    activityTypes: ['ENTITY_USAGE'],
    searchTerm: searchTerm,
    referredEntityType: EntityType.Secrets,
    endTime: useSecretRuntimeUsageTimeData().currentDate.valueOf(),
    startTime: useSecretRuntimeUsageTimeData().threeMonthsBack.valueOf()
  }
}

export const getOnlyValueTypeArray = (items: MultiSelectOption[]): string[] => {
  const valueItems = items.map(item => {
    return item.value
  })
  return valueItems as string[]
}

export const routeToEntityUrl = (targetUrl: string): void => {
  let baseUrl = getWindowLocationUrl()
  if (baseUrl.endsWith('/')) {
    baseUrl = baseUrl.substring(0, baseUrl.length - 1)
  }
  window.open(`${baseUrl}${targetUrl}`, '_blank')
}

export const filterData = (
  filterType: 'entity' | 'scope',
  items: MultiSelectOption[]
): {
  referredByEntityType?: string[]
  scopeFilter?: string[]
} => {
  const itemsWithOnlyValueType = getOnlyValueTypeArray(items)

  if (filterType === 'entity')
    return {
      referredByEntityType: [...itemsWithOnlyValueType]
    }
  if (filterType === 'scope')
    return {
      scopeFilter: [...itemsWithOnlyValueType]
    }
  else return {}
}
