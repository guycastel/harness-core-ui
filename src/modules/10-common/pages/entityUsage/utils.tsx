/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */
import React, { useMemo } from 'react'
import { CellProps, Renderer } from 'react-table'
import { Link, useParams } from 'react-router-dom'
import { MultiSelectOption, Text } from '@harness/uicore'
import { Color } from '@harness/design-system'
import { defaultTo, isEmpty } from 'lodash-es'
import { ExecutionPathProps, ProjectPathProps, SecretsPathProps } from '@common/interfaces/RouteInterfaces'
import { Activity, ListActivitiesQueryParams } from 'services/cd-ng'
import { StringKeys } from 'framework/strings'
import { COMMON_PAGE_SIZE_OPTIONS } from '@modules/10-common/constants/Pagination'
import { getWindowLocationUrl } from 'framework/utils/WindowLocation'
import routes from '@common/RouteDefinitions'
import { EntityType } from './EntityConstants'

export const pageSize = COMMON_PAGE_SIZE_OPTIONS[1]

enum FilterScope {
  ACCOUNT = 'account',
  ORG = 'org',
  PROJECT = 'project'
}

enum UsageType {
  PipelineExecution = 'Pipeline Execution',
  TestConnection = 'Test Connection'
}
export const getScopeSelectOptions = (getString: (key: StringKeys) => string): MultiSelectOption[] => {
  const scopeList = [
    { label: getString('account'), value: FilterScope.ACCOUNT },
    { label: getString('orgLabel'), value: FilterScope.ORG },
    { label: getString('projectLabel'), value: FilterScope.PROJECT }
  ]
  return scopeList
}

export const useSecretRuntimeUsageTimeData = (): { threeMonthsBack: Date; currentDate: number } => {
  return useMemo(() => {
    const threeMonthsBack = new Date()
    threeMonthsBack.setMonth(threeMonthsBack.getMonth() - 3)
    const currentDate = Date.now()

    return {
      threeMonthsBack: threeMonthsBack,
      currentDate: currentDate
    }
  }, [])
}

export const useGetSecretRuntimeUsageQueryParams = (
  page: number,
  searchTerm: string | undefined,
  pageContentSize?: number
): ListActivitiesQueryParams => {
  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
  const { secretId } = useParams<SecretsPathProps>()
  return {
    accountIdentifier: accountId,
    projectIdentifier: projectIdentifier,
    orgIdentifier: orgIdentifier,
    identifier: secretId,
    pageIndex: page,
    pageSize: pageContentSize || pageSize,
    activityTypes: ['ENTITY_USAGE'],
    searchTerm: searchTerm,
    referredEntityType: EntityType.Secrets,
    endTime: useSecretRuntimeUsageTimeData().currentDate.valueOf(),
    startTime: useSecretRuntimeUsageTimeData().threeMonthsBack.valueOf()
  }
}

export const getOnlyValueTypeArray = (items: MultiSelectOption[]): string[] => {
  const valueItems = items.map(item => {
    return item.value
  })
  return valueItems as string[]
}

export const routeToEntityUrl = (targetUrl: string): void => {
  let baseUrl = getWindowLocationUrl()
  if (baseUrl.endsWith('/')) {
    baseUrl = baseUrl.substring(0, baseUrl.length - 1)
  }
  window.open(`${baseUrl}${targetUrl}`, '_blank')
}

export const filterData = (
  filterType: 'entity' | 'scope',
  items: MultiSelectOption[]
): {
  referredByEntityType?: string[]
  scopeFilter?: string[]
} => {
  const itemsWithOnlyValueType = getOnlyValueTypeArray(items)

  if (filterType === 'entity')
    return {
      referredByEntityType: [...itemsWithOnlyValueType]
    }
  if (filterType === 'scope')
    return {
      scopeFilter: [...itemsWithOnlyValueType]
    }
  else return {}
}

export const usageTypeToLabelMap = (usageDetail: any): Renderer<CellProps<Activity>> => {
  const usageType: UsageType = !isEmpty(usageDetail.usagetype) ? usageDetail.usagetype : ''
  switch (usageType) {
    case UsageType.PipelineExecution: {
      const pipelineExecutionUsageData = usageDetail?.pipelineexecutionusagedata

      const {
        identifier: pipelineIdentifier,
        accountidentifier: accountId,
        orgidentifier: orgIdentifier,
        projectidentifier: projectIdentifier
      } = defaultTo(usageDetail?.identifierref, {})

      const planExecutionId = pipelineExecutionUsageData?.planexecutionid
      const stageExecutionId = pipelineExecutionUsageData?.stageexecutionid
      const source: ExecutionPathProps['source'] = pipelineIdentifier ? 'executions' : 'deployments'

      return (
        <Link
          to={routes.toExecutionPipelineView({
            orgIdentifier,
            projectIdentifier,
            accountId,
            source,
            executionIdentifier: planExecutionId || '-1',
            stageExecId: stageExecutionId || '-1',
            pipelineIdentifier: pipelineIdentifier || '-1'
          })}
          target="_blank"
        >
          {usageType}
        </Link>
      )
    }
    case UsageType.TestConnection:
      return (
        <Text color={Color.GREY_400} lineClamp={1}>
          {usageType}
        </Text>
      )
    default:
      return ''
  }
}
