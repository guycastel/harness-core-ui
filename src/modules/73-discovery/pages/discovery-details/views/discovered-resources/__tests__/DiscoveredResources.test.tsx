/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { act, fireEvent, screen, render, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import * as servicediscovery from 'services/servicediscovery'
import { TestWrapper } from '@common/utils/testUtils'
import DiscoveredResources from '../DiscoveredResources'
import { mockNamespaces, mockConnections } from '../mocks'

jest.mock('services/servicediscovery', () => ({
  useListDiscoveredServiceConnection: jest.fn().mockImplementation(() => {
    return { data: mockConnections, refetch: jest.fn(), error: null, loading: false }
  }),
  useListNamespace: jest.fn().mockImplementation(() => {
    return { data: mockNamespaces, refetch: jest.fn(), error: null, loading: false }
  })
}))

jest.mock('../graphView/DiscoveredResourcesGraph', () => ({
  ...jest.requireActual('../graphView/DiscoveredResourcesGraph'),
  __esModule: true,
  default: () => {
    return <div className={'discoveredResourcesGraph'}>Discovered Resources Graph</div>
  }
}))
jest.mock('../tableView/DiscoveredServices', () => ({
  ...jest.requireActual('../tableView/DiscoveredServices'),
  __esModule: true,
  default: () => {
    return <div className={'discoveredServices'}>Discovered Services</div>
  }
}))

describe('DiscoveredResources', () => {
  test('render component', async () => {
    const { container } = render(
      <TestWrapper>
        <DiscoveredResources />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('subheader components should be present', async () => {
    const { container, getByTestId, getByPlaceholderText } = render(
      <TestWrapper>
        <DiscoveredResources />
      </TestWrapper>
    )

    const dropdown = getByTestId('namespace')
    act(() => {
      fireEvent.click(dropdown)
    })

    const listItem = container.getElementsByClassName('DropDown--menuItem')[0]
    act(() => {
      fireEvent.click(listItem)
    })
    expect(listItem.textContent).toBe('All')

    const query = 'test abc'
    const searchInput = getByPlaceholderText('discovery.searchService') as HTMLInputElement
    expect(searchInput).not.toBe(null)
    if (!searchInput) {
      throw Error('no search input')
    }
    await userEvent.type(searchInput, query)
    await waitFor(() => expect(searchInput?.value).toBe(query))
  })

  test('Toggle between grid and list view', async () => {
    const { getByText } = render(
      <TestWrapper>
        <DiscoveredResources />
      </TestWrapper>
    )

    const gridToggle = screen.getByTestId('grid-view')
    await waitFor(() => expect(gridToggle).toBeInTheDocument())
    const listToggle = screen.getByTestId('list-view')
    await waitFor(() => expect(listToggle).toBeInTheDocument())

    fireEvent.click(listToggle!)
    expect(getByText('Discovered Services')).toBeInTheDocument()
    fireEvent.click(gridToggle!)
    expect(getByText('Discovered Resources Graph')).toBeInTheDocument()
  })

  test('when namespace data is undefined', async () => {
    jest.spyOn(servicediscovery, 'useListNamespace').mockImplementation((): any => {
      return {
        data: undefined,
        loading: false,
        error: {
          message: 'some error'
        }
      }
    })
    const { container } = render(
      <TestWrapper>
        <DiscoveredResources />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
