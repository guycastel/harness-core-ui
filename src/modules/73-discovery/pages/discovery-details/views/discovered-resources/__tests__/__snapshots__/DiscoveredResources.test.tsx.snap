// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscoveredResources render component 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
      data-testid="page-subheader"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
        style="gap: 8px; width: 100%;"
      >
        <div
          class="bp3-popover-wrapper DropDown--main"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                data-testid="namespace"
                style="width: 160px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                    common.namespace
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="bp3-input-group ui-search-box ExpandingSearchInput--main   ExpandingSearchInput--alwaysExpanded ExpandingSearchInput--light"
          data-name=""
          style="width: 232px; max-width: 232px;"
        >
          <span
            class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input  ExpandingSearchInput--filter"
            dir="auto"
            placeholder="discovery.searchService"
            style="padding-right: 38px;"
            type="search"
            value=""
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--gridButton StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
            data-testid="grid-view"
            data-tooltip-id="grid-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-grid-view StyledProps--main"
              icon="grid-view"
            >
              <svg
                data-icon="grid-view"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  grid-view
                </desc>
                <path
                  d="M0 1v6h7V0H1C.45 0 0 .45 0 1zm0 14c0 .55.45 1 1 1h6V9H0v6zM15 0H9v7h7V1c0-.55-.45-1-1-1zM9 16h6c.55 0 1-.45 1-1V9H9v7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--listUnselected GridListToggle--listButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
            data-testid="list-view"
            data-tooltip-id="list-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-list StyledProps--main"
              icon="list"
            >
              <svg
                data-icon="list"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  list
                </desc>
                <path
                  d="M1 3h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 10H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
    >
      <div
        class="discoveredResourcesGraph"
      >
        Discovered Resources Graph
      </div>
    </div>
  </div>
</div>
`;

exports[`DiscoveredResources when namespace data is undefined 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
      data-testid="page-subheader"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex"
        style="gap: 8px; width: 100%;"
      >
        <div
          class="bp3-popover-wrapper DropDown--main"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                data-testid="namespace"
                style="width: 160px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                    common.namespace
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="bp3-input-group ui-search-box ExpandingSearchInput--main   ExpandingSearchInput--alwaysExpanded ExpandingSearchInput--light"
          data-name=""
          style="width: 232px; max-width: 232px;"
        >
          <span
            class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input  ExpandingSearchInput--filter"
            dir="auto"
            placeholder="discovery.searchService"
            style="padding-right: 38px;"
            type="search"
            value=""
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--gridButton StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
            data-testid="grid-view"
            data-tooltip-id="grid-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-grid-view StyledProps--main"
              icon="grid-view"
            >
              <svg
                data-icon="grid-view"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  grid-view
                </desc>
                <path
                  d="M0 1v6h7V0H1C.45 0 0 .45 0 1zm0 14c0 .55.45 1 1 1h6V9H0v6zM15 0H9v7h7V1c0-.55-.45-1-1-1zM9 16h6c.55 0 1-.45 1-1V9H9v7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main GridListToggle--listUnselected GridListToggle--listButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
            data-testid="list-view"
            data-tooltip-id="list-view"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-list StyledProps--main"
              icon="list"
            >
              <svg
                data-icon="list"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  list
                </desc>
                <path
                  d="M1 3h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 10H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
    >
      <div
        class="discoveredResourcesGraph"
      >
        Discovered Resources Graph
      </div>
    </div>
  </div>
</div>
`;
